pylint/lint/pylinter.py

```diff
@@ -250,6 +261,68 @@ def _load_reporter_by_class(reporter_class: str) -> type[BaseReporter]:
 }
 
 
+def _handle_force_color_no_color(reporter: list[reporters.BaseReporter]) -> None:
+    """
+    Check ``NO_COLOR``, ``FORCE_COLOR``, ``PY_COLOR`` and modify the reporter list
+    accordingly.
+
+    Rules are presented in this table:
+    +--------------+---------------+-----------------+------------------------------------------------------------+
+    | `NO_COLOR`   | `FORCE_COLOR` | `output-format` | Behavior                                                   |
+    +==============+===============+=================+============================================================+
+    | `bool: True` | `bool: True`  | colorized       | not colorized + warnings (override + inconsistent env var) |
```

Comment:
@Pierre-Sassoulas opinion? (**strictly for the `env` C0402 annotation**)

There are "other" `env` references, like e.g. https://github.com/pylint-dev/pylint/blob/14b242f66d3c8f0e3f27b05f948ae9005052e8c0/pylint/checkers/stdlib.py#L895-L932

Reply:
Fixed by adding `env` to our `custom_dict` of allowed words :)